packages/jest-core/src/collectHandles.ts

```diff
@@ -83,16 +83,7 @@ export default function collectHandles(): HandleCollectionResult {
       // Skip resources that should not generally prevent the process from
       // exiting, not last a meaningfully long time, or otherwise shouldn't be
       // tracked.
-      if (
-        type === 'PROMISE' ||
-        type === 'TIMERWRAP' ||
-        type === 'ELDHISTOGRAM' ||
-        type === 'PerformanceObserver' ||
-        type === 'RANDOMBYTESREQUEST' ||
-        type === 'DNSCHANNEL' ||
-        type === 'ZLIB' ||
-        type === 'SIGNREQUEST'
-      ) {
+      if (type === 'PROMISE') {
```

Comment:
Only `TIMERWRAP` was added before jest repo was migrated to ts, it's a bit troublesome to find commit records, I don't have confirmation.
For other types, I have checked the submitted PRs and tests, and they are theoretically safe.
Anyway, this is in a major release and can be safely tested in a beta release.

Reply:
include `PerformanceObserver` to fix CI?

Reply:
Let me try to destroy it. https://github.com/facebook/jest/pull/13417#issuecomment-1272705013

Reply:
Original issue with a reproduction: https://github.com/facebook/jest/issues/11051

If node doesn't require a `disconnect` to exit cleanly, we shouldn't report it

Reply:
We can probably add

```js
const client = require("prom-client");
client.collectDefaultMetrics();
```

as an integration test

Reply:
I'm not really sure what `detectOpenHandles` is for now, technically most of the object types removed by this PR should not block process exit, but it seems that some users use this to detect memory leaks/object leaks. Would love to hear your opinion.

Reply:
It's used to detect handles keeping the process running after tests complete (such as a rogue server, `setInterval` (without `unref()` or `clearInterval`)) etc.. It shouldn't be used for memory leaks per se (although open handles indicate a missing cleanup of some sorts), but since it forces GC runs it oftens helps with memory usage

Reply:
essentially to find out why https://jestjs.io/docs/cli#--forceexit would be needed